### ISSUE TITLE
Intermittant Crash blit_to_string

### ISSUE THREAD

**Issue by mintsoft** [bcpierce00/unison#267]
Hiya,

We run Unison a few hundred times a day to sync up some servers, we find that occasionally (a few times per week) that Unison will crash (exit code 3) with:
```
Unison failed: Invalid argument: Bytearray.blit_to_string
Raised at file "./lwt/lwt.ml", line 135, characters 6-13
Called from file "list.ml", line 100, characters 12-15
Called from file "./lwt/lwt.ml", line 31, characters 2-37
Called from file "list.ml", line 100, characters 12-15
Called from file "./lwt/win/lwt_unix_impl.ml", line 243, characters 6-51
Called from file "./remote.ml", line 1483, characters 11-57
Called from file "./ubase/util.ml", line 170, characters 6-9
Called from file "./main.ml", line 131, characters 4-8
```

If you re-run the same command, it then succeeds, so it feels like a racecondition or similiar.

The commands we use are, on the remote:
`unison.exe -socket 874`

and on the local machine: 
`unison.exe -force "D:\SourceDirectory -batch -log=false -times -silent -killserver -halfduplex -ignorearchives "D:\SourceDirectory" "socket://remoteServer:874/C:\DestinationDirectory" -ignore "Name .svn" -ignore "Name .git" -ignore "Name .idea" -ignore "Name package.json" -ignore "Name node_modules" -ignore "Name tests" -ignore "Name gulpfile.js" -ignore "Name Gruntfile.js" -ignore "Name composer.phar";`

We're using 2.51.2 from: https://github.com/bcpierce00/unison/releases/download/v2.51.2/unison-windows-2.51.2-text.zip

How do we gather more information which could be useful?

**Comment by gdt** [comment]
Not sure anything is fixed, but it would be nice to test with 2.51.3.

**Comment by gdt** [comment]
timeout; feel free to reopen with 2.51.5 reproduction
